src/nrepl/config.clje

```diff
@@ -0,0 +1,61 @@
+(ns nrepl.config
+  "Server configuration utilities.
+  Some server options can be configured via configuration
+  files (local or global).  This namespace provides
+  convenient API to work with them.
+
+  The config resolution algorithm is the following:
+  The global config file .nrepl/nrepl.edn is merged with
+  any local config file (.nrepl.edn) if present.
+  The values in the local config file take precedence."
+  {:author "Bozhidar Batsov"
+   :added  "0.5.0"}
+  (:require
+   [clojure.erlang.io :as io]
+   [clojure.edn :as edn]))
+
+(def ^:private home-dir
+  "The user's home directory."
+  (os/getenv #erl"HOME"))
+
+(def ^:private separator
+  "MB: File.separator
+  This should be taken from File.separator, but I couldn't find it."
+  "/")
+
+(def config-dir
+  "nREPL's configuration directory.
+  By default it's ~/.nrepl, but this can be overridden
+  with the NREPL_CONFIG_DIR env variable."
+  (or (os/getenv #erl"NREPL_CONFIG_DIR")
+      (str home-dir separator ".nrepl")))
+
+(def config-file
+  "nREPL's config file."
+  (str config-dir separator "nrepl.edn"))
+
+(defn- load-edn
+  "Load edn from an io/reader source (filename or io/resource)."
+  [source]
+  (with-open [r (io/reader source)]
+    (edn/read (erlang.io.PushbackReader. r))))
+
+(defn- load-config
+  "Load the configuration file identified by `filename`.
+  Return its contents as EDN if the file exists,
+  or an empty map otherwise."
+  [filename]
+  (let [file (io/file filename)]
```

Comment:
There is no equivalent abstraction as Java's File object in Erlang. The `erlang.io.File` type is a wrapper around the [`file`](https://erlang.org/doc/man/file.html) module. Therefore it is necessary to check that a file exists with the weirdly named `filelib:is_regular/1` and then try to open it. 

Do you think it would be a good idea to wrap this behaviour in the `erlang.io.File`?

Reply:
I'm not sure. If there would be a function `exists` in filelib I'd say no.
But when there are two functions to call in different modules then that could be wrapped for convenience purposes.

Reply:
I think `exists` in Java checks if the file exists no matter if it's a regular file, a dir or a symlink and you have to check after if the file is a regular file or a dir.

We could do it similarly. `exists` could just be implemented by trying to open the file.